fastlane/lib/fastlane/actions/modify_services.rb

```diff
@@ -11,7 +11,7 @@ def self.run(params)
           services = params[:services]
 
           enabled_services = services.select { |_k, v| v == true || v.to_s != 'off' }.map { |k, v| [k, v == true || v.to_s == 'on' ? 'on' : v] }.to_h
-          disabled_services = services.reject { |_k, v| v == true || v.to_s == 'on' }.map { |k, v| [k, 'off'] }.to_h
+          disabled_services = services.select { |_k, v| v == false || v.to_s == 'off' }.map { |k, v| [k, 'off'] }.to_h
```

Comment:
👋 @janpio @martinkasa 

You can find the unit tests in [`modify_services_spec.rb`](https://github.com/fastlane/fastlane/blob/master/fastlane/spec/actions_specs/modify_services_spec.rb) file. 

Please correct me if I am wrong, but the problem is even though we set `data_protection: "complete"`, the service is enabled, but it gets disabled immediately after it. 

We can test it ☝️ in unit tests like so:

```Ruby
expect(Produce::Service).to receive(:disable) do |options, args|
    expect(options.associated_domains).to eq('off')
    expect(options.apple_pay).to eq('off')
    expect(options.multipath).to eq('off')
    expect(options.data_protection).to be_nil    // we do not expect data_protection being disabled
end
```

I ran the above test without your changes and the test failed:

```Bash
Failures:

  1) Fastlane Fastlane::FastFile modify_services sends enable and disable with strings, symbols, and booleans
     Failure/Error: expect(options.data_protection).to be_nil
     
       expected: nil
            got: "off"
```

Your changes fixes this 👏 👏 

```Bash
Fastlane
  Fastlane::FastFile
    modify_services
      sends enable and disable with strings, symbols, and booleans

Finished in 0.03534 seconds (files took 2.02 seconds to load)
1 example, 0 failures
```

Could you please modify unit tests according to the above suggestion? 🙏 

🙇